src/modules/pixi.rs

```diff
@@ -0,0 +1,140 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::pixi::PixiConfig;
+use crate::formatter::{StringFormatter, VersionFormatter};
+use crate::utils::get_command_string_output;
+
+/// Creates a module with the current Pixi environment
+///
+/// Will display the Pixi environment iff `$PIXI_ENVIRONMENT_NAME` is set.
+/// Will display the Pixi version iff pixi files are detected or `$PIXI_ENVIRONMENT_NAME` is set.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("pixi");
+    let config: PixiConfig = PixiConfig::try_load(module.config);
+
+    let pixi_environment_name = context.get_env("PIXI_ENVIRONMENT_NAME");
+    let is_pixi_project = pixi_environment_name.is_some()
+        || context
+            .try_begin_scan()?
+            .set_files(&config.detect_files)
+            .set_extensions(&config.detect_extensions)
+            .set_folders(&config.detect_folders)
+            .is_match();
+    if !is_pixi_project {
+        return None;
+    }
+
+    let pixi_environment_name = if !config.show_default_environment
+        && pixi_environment_name == Some("default".to_string())
+    {
+        None
+    } else {
+        pixi_environment_name
+    };
```

Comment:
I would prefer fully disabling the module if the default environment is detected like the conda module, at least by default.

Reply:
the way pixi and conda work in that regard are very different. in conda, the `base` environment is usually _always_ active when you are in your shell (imo stupid design decision but okay). there, it definitely makes sense to skip showing this in starship as it's present always anyway.

for pixi, on the other hand, you are not always inside of the `default` environment but only if you run `pixi shell` (which is equivalent to `conda activate <my-env>`).
for me, it makes more sense to see the `(default)` because it indicates me that i'm actually inside of a pixi shell (which isn't always the case).

that being said, if you feel very strongly about this, i can also set the default behavior to `show_default_environment: false`

Reply:
Sorry for not being clearer, I was referring to using `return None` instead of only hiding the environment name. Though, if base envs are actively enabled the current behaviour could be fine, even if the option isn't quite consistent with the conda module.

Reply:
ah okay but i still think that `return None` would be a bit weird here:

if you are not in a pixi shell but in a pixi project (i.e. `pixi.toml` exists), you will still get `🧚 v0.45.0` displayed. once you would go into the `pixi shell`, you would get nothing displayed anymore.

Reply:
> if base envs are actively enabled the current behaviour could be fine, even if the option isn't quite consistent with the conda module

there is no "base env" in pixi and going inside of the `default` environment is still an active decision by the user so i think it makes sense that this isn't consistent with the conda module